distributed/scheduler.py

```diff
@@ -2819,15 +2829,15 @@ def is_rootish(self, ts: TaskState) -> bool:
         Root-ish tasks are part of a group that's much larger than the cluster,
         and have few or no dependencies.
         """
-        if ts.resource_restrictions or ts.worker_restrictions or ts.host_restrictions:
+        if (
+            ts.resource_restrictions
+            or ts.worker_restrictions
+            or ts.host_restrictions
+            or ts.actor
+        ):
             return False
         tg = ts.group
-        # TODO short-circuit to True if `not ts.dependencies`?
-        return (
-            len(tg) > self.total_nthreads * 2
-            and len(tg.dependencies) < 5
-            and sum(map(len, tg.dependencies)) < 5
-        )
+        return len(tg.dependencies) < 5 and sum(map(len, tg.dependencies)) < 5
```

Comment:
I'm concerned about losing the `len(tg) > self.total_nthreads` check. That means that, for example, task `c` would now be considered root-ish:
```
  c
 / \
a   b
```

We don't want task `c` to be queued, for reasons described in the comment above. `c` isn't equally good to run on any worker—it should only be run on the workers that hold `a` or `b`. It's better for `c` run on the right worker, but maybe have to wait, than run on whichever wrong worker has a task slot open first.

The point of the `total_nthreads` check is that it identifies fan-out situations, where we go from dependencies that are on very few workers to dependents that will be on every worker. The fact that the dependencies will be copied to every worker is what allows us to ignore them while scheduling, which is what allows us to queue the tasks, since in queueing, worker busyness must be the only important scheduling factor.

Reply:
Why did you add the check against `total_nthreads` to the `queuing_disabled` case, but not `queuing_enabled`? If we could add the check into `queuing_enabled` as well, then I think this would be fine. But I'm also not clear on what the benefit of all this change would be then—the definition still isn't "static" in a practical sense, we'd just be moving the non-static decision to a different point in the code.